### ISSUE TITLE
[v43 Beta] Linking is disabled because the client scheme cannot be resolved.

### ISSUE THREAD

**Issue by Albert-Gao** [expo/expo-cli#3930]
### Summary

use Expo v43 Beta with dev-client, constantly get the error: `Linking is disabled because the client scheme cannot be resolved.`

![image](https://user-images.githubusercontent.com/18282328/138050192-37c37220-0971-4c0e-b9e6-987d71f1c459.png)

The app runs fine in the simulator, just the expo localhost portal is not showing up.

I tried: `EXPO_BETA=1 expo run:ios` and `EXPO_BETA=1 expo start --dev-client`, both gives the same error.

### Environment

```bash
  Expo CLI 4.12.5 environment info:
    System:
      OS: macOS 11.6
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.10.0 - ~/.nvm/versions/node/v16.10.0/bin/node
      Yarn: 1.22.15 - ~/.nvm/versions/node/v16.10.0/bin/yarn
      npm: 8.1.0 - ~/.nvm/versions/node/v16.10.0/bin/npm
      Watchman: 2021.06.07.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 15.0, DriverKit 20.4, macOS 11.3, tvOS 15.0, watchOS 8.0
      Android SDK:
        API Levels: 29, 31
        Build Tools: 29.0.2, 31.0.0
        System Images: android-31 | Google Play Intel x86 Atom_64
    IDEs:
      Android Studio: 4.2 AI-202.7660.26.42.7486908
      Xcode: 13.0/13A233 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^43.0.0-beta.3 => 43.0.0-beta.4
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.2 => 0.64.2
      react-native-web: 0.17.1 => 0.17.1
    npmGlobalPackages:
      expo-cli: 4.12.5
    Expo Workflow: bare
```

### Please specify your device/emulator/simulator platform, model and version

Simulator: iOS 15 iPhone 13

### Error output

![image](https://user-images.githubusercontent.com/18282328/138050944-cbc94e5d-0011-4447-be13-f56232b5da3e.png)

### Reproducible demo or steps to reproduce from a blank project

Private, can give access if needed

**Comment by Albert-Gao** [comment]
Found this link that produces this error message:

https://github.com/expo/expo-cli/blob/bf69269eee64b958574509176203d3fab290cbbd/packages/xdl/src/UrlUtils.ts#L68

What is the `ProjectSettings.readAsync(projectRoot);` here?

Is it `app.json`? If that is the case, my `schema` in `app.json` is a string:
![image](https://user-images.githubusercontent.com/18282328/138051510-4a2baf61-df8e-4d1b-97e0-8b798d8181a9.png)

also, `expo doctor` gives no error
![image](https://user-images.githubusercontent.com/18282328/138051578-4801f4c5-fc54-413c-96f5-e0b96937922a.png)

**Comment by kevinwcyu** [comment]
Downgrading `expo-cli` to `4.12.3` gets rid of the error for me.

> This error (if working as expected) is shown because there is no shared URI protocol for the iOS and Android apps. For example, if the iOS app can only open foo:// and the android app can only open bar:// then we cannot generate a cross-platform QR code, so we disable showing it.
> 
> To fix this, you can do one of the following:
> 
> managed: define a scheme property in the app.json then run prebuild again.
bare: npx uri-scheme add my-scheme then rebuild.

I already have a `scheme` property in my config and I ran `prebuild`. This added duplicate scheme entries in my AndroidManifest.xml and Info.plist.

When running `expo start --dev-client` on `4.12.3` I get

```
› Metro waiting on myscheme://expo-development-client/?url=http%3A%2F%2F192.168.11.130%3A8081
# ...  some more output
› Installing on iPhone 12
› Opening on iPhone 12 (com.bundle.identifier)
› Opening myscheme://expo-development-client/?url=http%3A%2F%2F127.0.0.1%3A8081 on iPhone 12
› Opening the iOS simulator, this might take a moment.
```

where `myscheme` is the scheme defined in my configuration, AndroidManifest.xml, and Info.plist files.

---

When running `expo start --dev-client` on `4.12.7` without any changes my configuration or other files I get

```
› Metro waiting on http://192.168.11.130:8081
# ... some more output
› Installing on iPhone 12
› Opening on iPhone 12 (com.bundle.identifier)
› Opening com.bundle.identifier on iPhone 12
iOS Bundling complete 1294ms
› Opening the iOS simulator, this might take a moment.
```

The app opens in iOS Simulator, but the Expo dev tools have the `Error loading DevTools No scheme specified for development client` error message.

Based on the output messages. It looks like `4.12.3` is able to get the scheme, but in more recent versions it's not getting it.

---

I'm on `expo-cli@4.12.7`, `expo@43.0.0` but saw this the beta versions as well. Some other notes. I had multiple schemes defined in my AndroidManifest.xml and Info.plist, which still worked with `expo-cli@4.12.3`. Also ran `expo doctor` which didn't report any issues.

**Comment by brentvatne** [comment]
thanks! this is resolved in expo-cli@4.12.8
